jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/PlatformParameters.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.gradle;
+
+import com.google.cloud.tools.jib.plugins.common.RawConfiguration.PlatformConfiguration;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+/** Configuration of a platform. */
+public class PlatformParameters implements PlatformConfiguration {
+  @Nullable String os;
+  @Nullable String architecture;
+
+  @Override
```

Comment:
Needs to add `@Input` as in [`ExtensionsParameters`](https://github.com/GoogleContainerTools/jib/blob/00dbfee080e66c238bd1f1168205cab9e96e11b7/jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ExtensionParameters.java#L34). This tells Gradle that `platform.os` is an input parameter.

And I think you may need setters. Please check if setting `platform.os = ...` in `build.gradle` actually works.